dbcon/mysql/ha_mcs_dml.cpp

```diff
@@ -904,7 +908,12 @@ int ha_mcs_impl_write_batch_row_(const uchar* buf, TABLE* table, cal_impl_if::ca
                             bitmap_set_bit(table->read_set, field->field_index);
                             String attribute;
                             field->val_str(&attribute);
-                            escape.assign((char*)buf, attribute.length());
+
+                            if (ci.utf8)
+                                escape.assign((char*)buf, attribute.length() * 3);
```

Comment:
I'm not sure about this change.  This String type (which is not std::string); does it return characters or bytes, any idea?